pwa-devdocs/src/technologies/basic-concepts/state-management/index.md

```diff
@@ -0,0 +1,203 @@
+
+Local state data is any data scoped within a component or its children components.
+This type of data is not shared with a component's parent or peer data.
+Otherwise, that data should be [lifted][].
+
+For example, a button component's disabled state is never used outside that component, so
+it is categorized as local state data.
+
+```jsx
+const MyButton = () => {
+    const [isDisabled, setIsDisabled] = useState(false);
+
+    const handleClick= useCallback(() => setIsDisabled(true),[])
+
+    return <button onClick={handleClick} disabled={isDisabled}>Click me!</button>;
+}
+```
+
+Global state data is any data made available to components in the entire application.
+Components that depend on a global state value subscribe to changes for that value and re-render themselves.
+Most components do not depend on the entire global state.
+Instead, a component only uses small pieces of the entire global state.
+
+Shopping cart data is an example of global state data that components in different levels of the application use and modify.
+
+## Common state management technologies
+
+There are many libraries and framework features that implement state management.
+This section describes two such technologies used in the PWA Studio project.
+
+### Redux
+
+[Redux][] is a state management design pattern and library.
+It promotes the idea of a global object tree that contains the state of the whole application.
+This object is known as a [store][].
+
+The store is a read-only object, which can only be updated by dispatching a [reducer][] function.
+Reducer functions accept the current state and an [action][] object as parameters and returns the next state.
+
+Application components are able to [dispatch][] various actions to update the state.
+Components can also [subscribe][] to state changes to update their appearance or behavior.
+
+Early versions of PWA Studio used the Redux library directly as the primary mechanism for managing application state,
+and the Redux pattern can be seen in hooks such as [`useQueryResult()`][] and [`useRestResponse()`][].
+
+Currently, PWA Studio abstracts away its Redux implementation details using Peregrine hooks and context providers.
+This opens up the possibility of the project replacing Redux in Peregrine with another state management library without breaking state dependent components, such as those in Venia.
+
+PWA Studio allows you to customize reducers and enhancers.
+The following example uses `combineReducers()` to combine the default Peregrine reducers with custom reducers specific to the project and uses the combined reducers when creating the Redux store.
+
+```jsx
+// Example src/store.js file
+
+import { combineReducers, createStore } from 'redux';
+import { enhancer, reducers } from '@magento/peregrine';
+
+import myReducers from './lib/reducers';
+
+// You can add your own reducers here and combine them with the Peregrine exports.
+const rootReducer = combineReducers([...reducers, ...myReducers]);
```

Comment:
Sorry, it's a single object as the param not an array. Spreading will merge reducers into a single object

 ```suggestion
const rootReducer = combineReducers({ ...reducers, ...myReducers });
```